main.go

```diff
@@ -219,28 +226,60 @@ func buildProxy(host, bind *url.URL, tokenSource oauth2.TokenSource) *httputil.R
 	return proxy
 }
 
-// findTokenSource fetches the reusable/cached oauth2 token source. If t is
-// provided, that token is used as a static value. Othwerise, this attempts to
-// get the renewable token from the environment (including via Application
-// Default Credentials).
-func findTokenSource(t string) (oauth2.TokenSource, error) {
+// findTokenSource fetches the reusable/cached oauth2 token source. If rawToken
+// is provided, that token is used as a static value and the audience parameter
+// is ignored. Othwerise, this attempts to get the renewable token from the
+// environment (via Application Default Credentials).
+func findTokenSource(ctx context.Context, rawToken, audience string) (oauth2.TokenSource, error) {
 	// Prefer supplied value, usually from the flag.
-	if t != "" {
-		token := new(oauth2.Token).WithExtra(map[string]interface{}{
-			"id_token": t,
-		})
+	if rawToken != "" {
+		token := &oauth2.Token{AccessToken: rawToken}
 		return oauth2.StaticTokenSource(token), nil
 	}
 
-	// Try and find the default token from ADC.
-	ctx := context.Background()
-	tokenSource, err := google.DefaultTokenSource(ctx, cloudPlatformScope)
+	// Try to use the idtoken package, which will use the metadata service.
+	// However, the idtoken package does not work with gcloud, so we need to
+	// handle that case by falling back to default ADC. However, the default ADC
+	// has a token at a different path, so we construct a custom token source for
+	// this edge case.
+	tokenSource, err := idtoken.NewTokenSource(ctx, audience)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get default token source: %w", err)
+		if !strings.Contains(err.Error(), "credential must be service_account") {
```

Comment:
How did you get this error message?

Reply:
https://github.com/googleapis/google-api-go-client/blob/3f4d679d4dae5de7ce124a7db68ef5147c9a3fa2/idtoken/idtoken.go#L140

Reply:
Could you add a comment for what situation we will get this. IIUC, in GCE instance, service account is used. If not service account, fall back to use gcloud's ADC.

Reply:
Added a comment. We would enter this clause under any other error conditions (metadata service comms issue, gcloud file permissions error, metadata concealment, etc).